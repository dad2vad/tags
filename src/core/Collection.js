class Collection {
  constructor(data = []) {
    this.data = data;
    this.count = data.length;
  }

  /***
   * Check if Collection has give item.
   *
   * @param item
   * @returns {boolean}
   */
  has(item) {
    return this.data.indexOf(item) > -1;
  }

  /**
   * Add unique item to Collection.
   *
   * @param item
   * @returns {Collection}
   */
  add(item) {
    if (!this.has(item)) {
      this.count = this.count + 1;
      this.data.push(item);
    }
    return this;
  }

  /***
   * Remove item from Collection.
   *
   * @param item
   * @returns {Collection}
   */
  remove(item) {
    if (this.has(item)) {
      this.data.splice(this.data.indexOf(item), 1);
      this.count = this.count - 1;
    }

    return this;
  }

  /***
   * Remove first item from Collection.
   *
   * @returns {Collection}
   */
  removeFirst() {
    if (this.count > 0) {
      this.data.shift();
      this.count = this.count - 1;
    }

    return this;
  }

  /***
   * Return total size of Collection items.
   *
   * @returns {*}
   */
  size() {
    return this.count;
  }

  /**
   * Return all elements from Collection.
   *
   * @returns {Array|*}
   */
  all() {
    return this.data;
  }
}

export default Collection;
