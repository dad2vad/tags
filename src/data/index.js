import { AmazeUI } from './frameworks/AmazeUI';
import { Axiom } from './frameworks/Axiom';
import { BlazeCSS } from './frameworks/BlazeCSS';
import { Bootstrap3, Bootstrap4 } from './frameworks/Bootstrap';
import { Bulma } from './frameworks/Bulma';
import { Foundation5, Foundation6 } from './frameworks/Foundation';
import { Gumby } from './frameworks/Gumby';
import { Ink } from './frameworks/Ink';
import { Kube } from './frameworks/Kube';
import { Materialize } from './frameworks/Materialize';
import { MetroUI } from './frameworks/MetroUI';
import { Mini } from './frameworks/Mini';
import { MUI } from './frameworks/MUI';
import { OfficeUI } from './frameworks/OfficeUI';
import { UIkit2, UIkit3 } from './frameworks/UIkit';
import { Pure } from './frameworks/Pure';
import { Primer } from './frameworks/Primer';
import { SemanticUI } from './frameworks/SemanticUI';
import { Skeleton } from './frameworks/Skeleton';
import { Spectre } from './frameworks/Spectre';

import Accordion from './components/Accordion';
import Alert from './components/Alert';
import Avatar from './components/Avatar';
import Badge from './components/Badge';
import Breadcrumbs from './components/Breadcrumbs';
import Button from './components/Button';
import Card from './components/Card';
import Dropdown from './components/Dropdown';
import FlexGrid from './components/FlexGrid';
import Form from './components/Form';
import Grid from './components/Grid';
import Icons from './components/Icons';
import Label from './components/Label';
import List from './components/List';
import Loading from './components/Loading';
import Media from './components/Media';
import Menu from './components/Menu';
import Modal from './components/Modal';
import Navbar from './components/Navbar';
import OffCanvas from './components/OffCanvas';
import Pagination from './components/Pagination';
import Panel from './components/Panel';
import Popover from './components/Popover';
import Progress from './components/Progress';
import Table from './components/Table';
import Tabs from './components/Tabs';
import Tags from './components/Tags';
import Thumbnail from './components/Thumbnail';
import Tooltip from './components/Tooltip';
import Typography from './components/Typography';
import Utilities from './components/Utilities';

export default {
  frameworks: [
    AmazeUI,
    Axiom,
    BlazeCSS,
    Bootstrap3,
    Bootstrap4,
    Bulma,
    Foundation5,
    Foundation6,
    Gumby,
    Ink,
    Kube,
    Materialize,
    MetroUI,
    Mini,
    MUI,
    OfficeUI,
    UIkit2,
    UIkit3,
    Pure,
    Primer,
    SemanticUI,
    Skeleton,
    Spectre,
  ],
  components: [
    Accordion,
    Alert,
    Avatar,
    Badge,
    Breadcrumbs,
    Button,
    Card,
    Dropdown,
    FlexGrid,
    Form,
    Grid,
    Icons,
    Label,
    List,
    Loading,
    Media,
    Menu,
    Modal,
    Navbar,
    OffCanvas,
    Pagination,
    Panel,
    Popover,
    Progress,
    Table,
    Tabs,
    Tags,
    Thumbnail,
    Tooltip,
    Typography,
    Utilities,
  ],
};
