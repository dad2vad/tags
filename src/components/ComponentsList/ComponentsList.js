import React from 'react';
import PropTypes from 'prop-types';
import DemoButton from '../DemoButton';
import NotAvailable from '../NotAvailable';
import Collection from '../../core/Collection';

const ComponentsList = ({ frameworks, visible, components }) => {
  function renderHeader() {
    return (
      <thead>
        <tr>
          <th className="is-first">
            <span className="is-dark">Component Name</span>
          </th>
          {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
            return (
              <th key={framework.slug}>
                <span className="is-dark">{framework.name}</span>
              </th>
            );
          })}
        </tr>
      </thead>
    );
  }

  function renderFooter() {
    return (
      <tfoot>
        <tr>
          <th className="is-first">
            <span className="is-dark">Component Name</span>
          </th>
          {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
            return (
              <th key={framework.slug}>
                <span className="is-dark">{framework.name}</span>
              </th>
            );
          })}
        </tr>
      </tfoot>
    );
  }

  return (
    <div className="section">
      <div className="container is-x is-fullhd">
        <h2 className="title">List of Components</h2>
        <table className="table has-text-centered is-striped is-bordered">
          {renderHeader()}
          <tbody>
            {components.map(component => {
              return (
                <tr key={component.name.toLowerCase()}>
                  <td>
                    <span className="tag is-info is-rounded">{component.name}</span>
                  </td>

                  {frameworks.filter(framework => visible.has(framework.slug)).map(framework => {
                    const supportsComponent = component.supports[framework.slug];
                    return (
                      <td key={framework.slug}>
                        {supportsComponent && <DemoButton url={supportsComponent} />}

                        {!supportsComponent && <NotAvailable />}
                      </td>
                    );
                  })}
                </tr>
              );
            })}
          </tbody>
          {renderFooter()}
        </table>
      </div>
    </div>
  );
};

ComponentsList.propTypes = {
  frameworks: PropTypes.array.isRequired,
  visible: PropTypes.instanceOf(Collection).isRequired,
  components: PropTypes.array.isRequired,
};

export default ComponentsList;
