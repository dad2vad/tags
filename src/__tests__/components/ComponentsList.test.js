import React from 'react';
import { shallow } from 'enzyme';
import Collection from '../../core/Collection';
import data from '../../data';
import ComponentsList from '../../components/ComponentsList';

describe('Components List', () => {
  it('renders correctly', () => {
    const visible = new Collection(['bulma', 'ink']);
    const component = shallow(
      <ComponentsList components={data.components} frameworks={data.frameworks} visible={visible} />
    );
    expect(component).toMatchSnapshot();
  });
});
