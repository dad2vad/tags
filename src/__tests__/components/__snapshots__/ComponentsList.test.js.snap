// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components List renders correctly 1`] = `
<div
  className="section"
>
  <div
    className="container is-x is-fullhd"
  >
    <h2
      className="title"
    >
      List of Components
    </h2>
    <table
      className="table has-text-centered is-striped is-bordered"
    >
      <thead>
        <tr>
          <th
            className="is-first"
          >
            <span
              className="is-dark"
            >
              Component Name
            </span>
          </th>
          <th>
            <span
              className="is-dark"
            >
              Bulma
            </span>
          </th>
          <th>
            <span
              className="is-dark"
            >
              Ink Interface Kit
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Accordion
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/javascript-ui/#InkUIToggle_1"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Alert
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/notification/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/alerts/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Avatar
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Badge
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/tag/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/typography/#labels-and-badges"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Breadcrumbs
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/breadcrumb/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/navigation/#breadcrumbs"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Button
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/button/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/buttons/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Card
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/card/"
            />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Dropdown
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/dropdown/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/navigation/#dropdowns"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Flex Grid
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/grid/columns/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/flexbox/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Form
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/form/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/forms/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Grid
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/grid/columns/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/grid/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Icons
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/icon/"
            />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Label
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/tag/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/typography/#labels-and-badges"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              List
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Loading
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Media Object
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/media-object/"
            />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Menu
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/menu/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/navigation/#menus"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Modal
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/modal/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/javascript-ui/#InkUIModal_1"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Navbar
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/navbar/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/navigation/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Off-Canvas
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Pagination
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/pagination/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/navigation/#pagination"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Panel
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/panel/"
            />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Popover
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Progress
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/progress/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/javascript-ui/#InkUIProgressBar_1"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Table
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/table/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/tables/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Tabs
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/components/tabs/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/javascript-ui/#InkUITabs_1"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Tags
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <NotAvailable />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Thumbnail
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/image/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/images/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Tooltip
            </span>
          </td>
          <td>
            <NotAvailable />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/javascript-ui/#InkUITooltip_1"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Typography
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/elements/content/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/typography/"
            />
          </td>
        </tr>
        <tr>
          <td>
            <span
              className="tag is-info is-rounded"
            >
              Utilities
            </span>
          </td>
          <td>
            <DemoButton
              url="http://bulma.io/documentation/modifiers/helpers/"
            />
          </td>
          <td>
            <DemoButton
              url="http://ink.sapo.pt/ui-elements/helpers/"
            />
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <th
            className="is-first"
          >
            <span
              className="is-dark"
            >
              Component Name
            </span>
          </th>
          <th>
            <span
              className="is-dark"
            >
              Bulma
            </span>
          </th>
          <th>
            <span
              className="is-dark"
            >
              Ink Interface Kit
            </span>
          </th>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
